app/controllers/api/shops_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Api
+  class ShopsController < BaseController
+    respond_to :json
+    skip_authorization_check only: [:show, :closed_shops]
+
+    def show
+      enterprise = Enterprise.find_by_id(params[:id])
+
+      render text: Api::EnterpriseShopfrontSerializer.new(enterprise).to_json, status: :ok
+    end
+
+    def closed_shops
```

Comment:
IMO this should be a filter specified by URL query param for an `API::ShopsController#index` so we follow Rails REST conventions. But I want to discuss this in another PR and without any rush. This is awesome as it is.